app/src/main/java/org/mozilla/reference/browser/Components.kt

```diff
@@ -4,211 +4,24 @@
-        intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK
-        context.startActivity(intent)
-    }
-
-    private val menuToolbar by lazy {
-        val forward = BrowserMenuItemToolbar.Button(
-                mozilla.components.ui.icons.R.drawable.mozac_ic_forward,
-                iconTintColorResource = R.color.icons,
-                contentDescription = "Forward") {
-            sessionUseCases.goForward.invoke()
-        }
-
-        val refresh = BrowserMenuItemToolbar.Button(
-                mozilla.components.ui.icons.R.drawable.mozac_ic_refresh,
-                iconTintColorResource = R.color.icons,
-                contentDescription = "Refresh") {
-            sessionUseCases.reload.invoke()
-        }
-
-        val stop = BrowserMenuItemToolbar.Button(
-                mozilla.components.ui.icons.R.drawable.mozac_ic_stop,
-                iconTintColorResource = R.color.icons,
-                contentDescription = "Stop") {
-            sessionUseCases.stopLoading.invoke()
-        }
-
-        BrowserMenuItemToolbar(listOf(forward, refresh, stop))
-    }
-
-    // Tabs
-    val tabsUseCases: TabsUseCases by lazy { TabsUseCases(sessionManager) }
-
-    // TODO work around until we have https://github.com/mozilla-mobile/android-components/issues/1457
-    val removeSessions = {
-        sessionManager.all.filter { session ->
-            !session.private
-        }.forEach { session ->
-            sessionManager.remove(session)
-        }
-    }
-
-    val removePrivateSessions = {
-        sessionManager.all.filter { session ->
-            session.private
-        }.forEach { session ->
-            sessionManager.remove(session)
-        }
-    }
-
-    // Firefox Accounts
-    val firefoxAccountsIntegration: FirefoxAccountsIntegration by lazy {
-        FirefoxAccountsIntegration(context, tabsUseCases, firefoxSyncFeature)
-    }
-
-    val crashReporter: CrashReporter by lazy {
-
-        val sentryService = SentryService(
-            context,
-            SENTRY_TOKEN,
-            sendEventForNativeCrashes = true
-        )
-
-        val socorroService = MozillaSocorroService(context, "ReferenceBrowser")
-
-        CrashReporter(
-            services = listOf(sentryService, socorroService),
-            shouldPrompt = CrashReporter.Prompt.ALWAYS,
-            promptConfiguration = CrashReporter.PromptConfiguration(
-                appName = context.getString(R.string.app_name),
-                organizationName = "Mozilla"
-            ),
-            nonFatalCrashIntent = createNonFatalPendingIntent(),
-            enabled = true
-        )
-    }
-
-    private fun createNonFatalPendingIntent(): PendingIntent {
-        return getBroadcast(context, 0, Intent(NON_FATAL_CRASH_BROADCAST), 0)
-    }
+    val core by lazy { Core(context) }
+    val search by lazy { Search(context) }
+    val useCases by lazy { UseCases(context, core.sessionManager, search.searchEngineManager) }
```

Comment:
Over time we might find that dependency on a group (vs members of a group) is less cumbersome, but this is fine for now.

Reply:
Agreed, also. :)